data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java

```diff
@@ -42,4 +52,17 @@ public Object coercePrimaryTerminalNode(final TerminalNode node, final Event eve
         }
         throw new ExpressionCoercionException("Unable to cast " + obj.getClass().getName() + " into " + clazz.getName());
     }
+
+    private Object resolveJsonPointerValue(final String jsonPointer, final Event event) {
+        final Object value = event.get(jsonPointer, Object.class);
+        if (value == null) {
+            throw new ExpressionCoercionException("Unsupported type for value " + value);
+        } else if (literalTypes.contains(value.getClass())) {
+            return value;
+        } else if (value instanceof Double) {
```

Comment:
This approach would be easier to extend if we have other possible types.


```
private static final Map<Class, Function<Object, Object>> literalTypes;

static {
  literalTypes = new HashMap<>();
  literalTypes.put(String.class, Function.identity());
  literalTypes.put(Boolean.class, Function.identity());
  literalTypes.put(Integer.class, Function.identity());
  literalTypes.put(Float.class, Function.identity());
  literalTypes.put(Double.class, v -> ((Double) value).floatValue());
}
```

And here:
```
else if (literalTypes.containsKey(value.getClass())) {
             return literalTypes.get(value.getClass()).apply(value);
```

But, this is all internal and later refactoring is light. So I don't have a strong opinion on it.